### ISSUE TITLE
Qt metapackage mutually incompatible with PyQt on Python 3.11

### ISSUE THREAD

**Issue by CAM-Gerlach** [conda-forge/qt-feedstock#234]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

When upgrading QtPy's CI matrix to test on Python 3.11 in spyder-ide/qtpy#392 , I've discovered an issue with the current pinning in this metapackage on Python 3.11, which @ccordoba12 recommended I report here, where it seems most proximate and solvable. Specifically, no version of the `qt` metapackage and `pyqt` are mutually compatible on Python 3.11, and running

```shell
conda create -n test-env python=3.11 pyqt qt
```

or equivalently

```shell
conda create -n test-env python=3.11 pyqt=5.15 qt=5.15
```

fails on all platforms with a dependency resolution error, which you can see the full output of on CI for each platform in e.g. [this run](https://github.com/spyder-ide/qtpy/actions/runs/3718737526/jobs/6307127310).

Specifically, I've traced the issue to following three incompatible constraints:

* The latest `pyqt` 5.15.7 is the earliest version supported on Python 3.11 (5.15.4 will not install)
* `pyqt` 5.15.7 requires `qt-main`, of which only 5.15.6 is mutually compatible with `pyqt` (5.15.4 is mutually incompatible, but will install on its own on Python 3.11, or with `pyqt` 5.15.7on Python <3.11)
* The latest version of the `qt` 5.15 meta-package is 5.15.4, which pins on `qt-main` and `qt-webengine` of the same point version

To note,

```shell
conda create -n test-env python=3.11 pyqt=5.15.7 qt-main=5.15.6 qt-webengine=5.15.4
```

does resolve. Also, 5.15.4 is the latest version of `qt-webengine`, though conda-forge/qt-webengine-feedstock#20 is open to update it to 5.15.7—though that wouldn't be in sync with the latest `qt-main` version, so I'm not sure if that can be pinned with `qt` unless `qt-main` is also updated to that version.

Presumably, at least one of the following need to happen:

* A new re-rendered build of the existing versions of the `qt` metapackage and the `qt-main` 5.15.4 packge, if conda-smithy re-rendering for Python 3.11 will fix it.
* A new release of the `qt` metapackage keeping the existing `qt-webengine` and upgrading `qt-main` to the working 5.15.6 version (but then the versions wouldn't match)
* Updating `qt-webengine` to 5.15.6 and releasing a new `qt` 5.15.6 metapackage with both deps pinned to that version 
* Merging conda-forge/qt-webengine-feedstock#20  for `qt-webengine` 5.15.7, and updating `qt-main` to 5.15.7, and finally releasing a new `qt` metapackage version accordingly
* Some other easier solution to the fundamental incompatibility?

However, you'd be the ones with the expertise and experience to determine the best course of action. Thanks!

### Installed packages

See the `Test PyQt` step of [each CI runs](https://github.com/spyder-ide/qtpy/actions/runs/3718737526/jobs/6307127310) for full command output, `conda list`, conflict information and other details for each platform.

In addition, the problem can be reproduced in a new clean environment by running

conda create -n test-env python=3.11 pyqt qt

or equivalently

```shell
conda create -n test-env python=3.11 pyqt=5.15 qt=5.15
```

### Environment info

See the `Conda Info` step of [each CI runs](https://github.com/spyder-ide/qtpy/actions/runs/3718737526/jobs/6307127310) for full output of `conda info` and other environment information for each platform.

**Comment by hmaarrfk** [comment]
I think I likely "fixed" this in https://github.com/conda-forge/qt-feedstock/pull/235

Though we really don't recommend depending on this package anymore.

Most of the effort has moved toward `qt-main`

**Comment by CAM-Gerlach** [comment]
Yup, I can confirm this is fixed by #235—the above install specs now resolve successfully, and our CIs work again without the complexity of separate code paths for installing different minor versions. We don't actually hard depend on the `qt` metapackage, and for Qt6 on Conda our CIs install `qt6-main`, so this shouldn't be an issue long term. Thanks!
